### PR TITLE
Adopt new api

### PR DESCRIPTION
**Pull request by myronkscott**
1.  Upgrade to the new tc-api.  This is a step toward not requiring implementation libraries in tc-core as dependencies.  

2. Add functionality to message tracker required for fixing data loss in 3 node clusters.

### CODE REVIEW

File: diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java [hunk 1]
```diff
@@ -93,23 +94,17 @@ public void test_generateMBean() {
...
   }
 
-  private void verifyMBean(TerracottaMBean mBean) {
+  private void verifyMBean(StandardMBean mBean) {
     assertThat(mBean, is(
         either(instanceOf(MyService.class))
             .or(instanceOf(MyService2.class))));
-    assertThat(mBean.getInterfaceClassName(), is(
-        either(equalTo(MyService.class.getName() + "MBean"))
-            .or(equalTo(MyService.class.getName() + "2MBean"))));
```
Comment (mathieucarbou) [inline diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java:109]: why these verifications have been removed ?

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/service/DynamicConfigServiceImpl.java [hunk 2]
```diff
@@ -342,7 +342,7 @@ public void stop(Duration delayInSeconds) {
     runAfterDelay(delayInSeconds, () -> {
       LOGGER.info("Stopping node");
       //TODO [DYNAMIC-CONFIG]: TDB-4942 - when upgrading to the new core version, use ZAP_AND_STOP instead
-      TCServerMain.getServer().stop(PlatformService.RestartMode.STOP_ONLY);
+      ServerEnv.getServer().stop();
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/service/DynamicConfigServiceImpl.java:345]: Now that we have the new API, could you please change to `ServerEnv.getServer().stop(ZAP);` and remove the TODO ?

File: dynamic-config/server/configuration-provider/pom.xml [hunk 3]
```diff
@@ -75,12 +75,23 @@
...
+    <dependency>
+      <groupId>org.terracotta</groupId>
+      <artifactId>standard-cluster-services</artifactId>
+      <scope>provided</scope>
+    </dependency>
     <dependency>
       <groupId>org.terracotta.internal</groupId>
-      <artifactId>dso-l2</artifactId>
+      <artifactId>common-spi</artifactId>
+      <version>${terracotta-core.version}</version>
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/pom.xml:86]: `<version>` should be moved in parent pom (and lib added in dependency management)

File: dynamic-config/server/configuration-provider/pom.xml [hunk 4]
```diff
@@ -75,12 +75,23 @@
...
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.terracotta.internal</groupId>
+      <artifactId>configuration-provider</artifactId>
+      <version>${terracotta-core.version}</version>
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/pom.xml:92]: `<version>` should be moved in parent pom (and lib added in dependency management)

File: diagnostic/service/pom.xml [hunk 5]
```diff
@@ -42,13 +42,14 @@
...
-      <artifactId>management</artifactId>
+      <groupId>org.terracotta</groupId>
+      <artifactId>standard-cluster-services</artifactId>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.terracotta.internal</groupId>
-      <artifactId>dso-l2</artifactId>
+      <artifactId>common-spi</artifactId>
+      <version>${terracotta-core.version}</version>
```
Comment (mathieucarbou) [inline diagnostic/service/pom.xml:52]: `<version>` should be moved in parent pom (and lib added in dependency management)

File: pom.xml [hunk 6]
```diff
@@ -111,12 +111,12 @@
       </dependency>
       <dependency>
         <groupId>org.terracotta.internal</groupId>
-        <artifactId>dso-l1</artifactId>
-        <version>${terracotta-core.version}</version>
+        <artifactId>common-spi</artifactId>
+        <scope>provided</scope>
```
Comment (mathieucarbou) [inline pom.xml:115]: - `<scope>` should be removed
- `<version>${terracotta-core.version}</version>` is missing
(to be consistent with how we handle deps in this project)

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java [hunk 7]
```diff
@@ -224,7 +225,7 @@ private void warnIfPreparedChange() {
 
   @SuppressWarnings({"unchecked", "rawtypes"})
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:226]: I think we do not need that now ?

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java [hunk 8]
```diff
@@ -179,16 +177,19 @@ public void sync(byte[] bytes) {
       Set<Require> requires;
       try {
         requires = dynamicConfigurationPassiveSync.sync(DynamicConfigSyncData.decode(bytes));
-      } catch (NomadException | RuntimeException e) {
-        // shutdown the server because of a unrecoverable error
-        throw new TCShutdownServerException("Shutdown because of sync failure: " + e.getMessage(), e);
+      } catch (NomadException n) {
+        throw new RuntimeException(n);
+      } catch (RuntimeException e) {
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:182]: can we remove this catch and eventually put the comment elsewhere if we really need to keep it ? this is the same behavior as for the rethrowing of the NomadException into a runtime, so explanation could be moved elsewhere, as to why we can rethrow as runtime.

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java [hunk 9]
```diff
@@ -179,16 +177,19 @@ public void sync(byte[] bytes) {
...
+        // reply on the server to stop based on uncaught exception
+        throw e;
       }
 
       if (requires.contains(RESTART_REQUIRED)) {
         if (requires.contains(ZAP_REQUIRED)) {
-          throw new ZapDirtyDbServerNodeException("Zapping server");
+          ServerEnv.getServer().warn("Zapping server");
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:189]: what is the difference between using this `warn` and using a logger ?

**Comment by mathieucarbou** [comment]
Thanks for the change @myronkscott ! This really cleans up the dependencies over internal stuff 👍

File: diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java (hunk 1 above)
Comment (myronkscott) [reply diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java:109]: `getInterfaceName()` is not part of StadardMBean signature.  I'll see if I can find a substitute

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/service/DynamicConfigServiceImpl.java (hunk 2 above)
Comment (myronkscott) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/service/DynamicConfigServiceImpl.java:345]: will do

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java (hunk 8 above)
Comment (myronkscott) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:182]: I'm not understanding this comment.

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java (hunk 9 above)
Comment (myronkscott) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:189]: Up to the server what to do with the warning.  Examples might be log or dump jfr etc.

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java (hunk 8 above)
Comment (mathieucarbou) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:182]: we could just simplify:
```java
      try {
        requires = dynamicConfigurationPassiveSync.sync(DynamicConfigSyncData.decode(bytes));
      } catch (NomadException n) {
        throw new RuntimeException(n);
      } 
```
If we **really** need to add a comment, we can put this comment before the try/catch explaining why throwing runtime is enough.

Comment (myronkscott) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java]: I see.  Makes sense.

File: diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java (hunk 1 above)
Comment (myronkscott) [reply diagnostic/service/src/test/java/org/terracotta/diagnostic/server/TerracottaMBeanGeneratorTest.java:109]: restored and changed to `mBean.getMBeanInterface().getName()`

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java (hunk 9 above)
Comment (mathieucarbou) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/DynamicConfigConfigurationProvider.java:187]: ok!

**Comment by mathieucarbou** [comment]
@saurabhagas : talked to @myronkscott : let's wait until we have finished the fixes required for XPC before merging this PR since adapting EE might be slower than expected.

**Comment by mathieucarbou** [comment]
@myronkscott @saurabhagas  :this PR can rebase and merged now.

**Comment by saurabhagas** [comment]
Needs a rebase

**Pull request merged**